week3/final/robot-text/robot.js

```diff
@@ -1,110 +1,122 @@
+    console.log(applesEaten + " apples were eaten!");
   }
+}
 
-  function move() {
-    let x = robot.x;
-    let y = robot.y;
+function move() {
+  let x = robot.x;
+  let y = robot.y;
+  appleEaten = false;
 
-    switch (robot.dir) {
-      case 'up':
-        y = y < board.length - 1 ? y + 1 : y;
-        break;
-      case 'down':
-        y = y > 0 ? y - 1 : y;
-        break;
-      case 'left':
-        x = x > 0 ? x - 1 : x;
-        break;
-      case 'right':
-        x = x < board[y].length - 1 ? x + 1 : x;
-        break;
-    }
+  switch (robot.dir) {
+    case 'up':
+      y = y < board.length - 1 ? y + 1 : y;
+      break;
+    case 'down':
+      y = y > 0 ? y - 1 : y;
+      break;
+    case 'left':
+      x = x > 0 ? x - 1 : x;
+      break;
+    case 'right':
+      x = x < board[y].length - 1 ? x + 1 : x;
+      break;
+  }
 
-    const cellContents = board[y][x];
+  const cellContents = board[y][x];
 
-    if (cellContents === '.' || cellContents === 'F') {
-      board[robot.y][robot.x] = trailIndicators[robot.dir];
-      robot.x = x;
-      robot.y = y;
-      board[y][x] = 'R';
-      if (cellContents === 'F') {
-        flagReached = true;
-      }
+  if (cellContents === '.' || cellContents === 'A' || cellContents === 'F'){
+    board[robot.y][robot.x] = trailIndicators[robot.dir];
+    robot.x = x;
+    robot.y = y;
+    board[y][x] = 'R';
+    if (cellContents === 'F') {
+      flagReached = true;
     }
-
-    moves += 1;
-    render();
+    if (cellContents === 'A') {
+      appleEaten = true;
+      applesEaten++;
+    }    
+    
   }
 
-  function turn(turnDirection) {
-    if (turnDirection !== 'left' && turnDirection !== 'right') {
-      console.log('ignoring invalid turn', turnDirection);
-    }
-    switch (robot.dir) {
-      case 'up':
-        robot.dir = turnDirection === 'left' ? 'left' : 'right';
-        break;
-      case 'down':
-        robot.dir = turnDirection === 'left' ? 'right' : 'left';
-        break;
-      case 'left':
-        robot.dir = turnDirection === 'left' ? 'down' : 'up';
-        break;
-      case 'right':
-        robot.dir = turnDirection === 'left' ? 'up' : 'down';
-        break;
-    }
-  }
 
+  moves += 1;
   render();
+}
+
+function turn(turnDirection) {
+  if (turnDirection !== 'left' && turnDirection !== 'right') {
+    console.log('ignoring invalid turn', turnDirection);
+  }
+  switch (robot.dir) {
+    case 'up':
+      robot.dir = turnDirection === 'left' ? 'left' : 'right';
+      break;
+    case 'down':
+      robot.dir = turnDirection === 'left' ? 'right' : 'left';
+      break;
+    case 'left':
+      robot.dir = turnDirection === 'left' ? 'down' : 'up';
+      break;
+    case 'right':
+      robot.dir = turnDirection === 'left' ? 'up' : 'down';
+      break;
+  }
+}
 
-  move();
-  turn('right');
-  move();
-  move();
-  move();
-  turn('left');
-  move();
-  move();
+render();
 
-})();
+turn('down');
```

Comment:
I guess this was a test with 'down'? When you submit a pull request make sure you take out all your debug code.

Reply:
Yes, I try to decode the concept behind the directions, it did not make sense. I did not notice the debug code was there. Wow I just notice what it meant.

Reply:
Yes, the `right` and `left` are relative turn directions. `up`, `down`, `left` and `right` are the absolute (compass) directions for the robot. Is this now your current understanding?

Reply:
Yes, that is my current understanding now. But the direction cannot change from → to ←? It can only turn to either left or right.

Reply:
Turn left or right twice in a row?

See the diagram in the [README](https://github.com/remarcmij/JavaScript-1/blob/master/week3/robot.md). You did see that, did you?

Reply:
Seeing the arrow keys in keyboard, as a 'confirmation bias' I had assumed to see a down or up turn. Now I understand it completely after seeing the README just now. I had seen it on your youtube recording.